doc/directory-structure.md

```diff
@@ -0,0 +1,183 @@
+* `Configure` - The main build configuration script
+* `config` - A wrapper around Configure for Unix platforms
+* `LICENSE.txt` - The Apache 2.0 license that applies to OpenSSL 3.0+
+* `CHANGES.md` - A comprehensive changelog
+* `NEWS.md` - A curated list of major changes in each release
+* `VERSION.dat` - Contains the current version information
+* `*.pc.in` and `configdata.pm.in` - Templates used during build configuration
+
+Various platform-specific notes are also at the root level:
+* `NOTES-*.md` - Platform-specific information (UNIX, WINDOWS, etc.)
+* `README-*.md` - Additional READMEs for specific components 
+  (ENGINES, FIPS, etc.)
+
+## Major Directories
+
+### apps/
+
+Contains the OpenSSL command-line application source code. 
+This is what you're using when you run the `openssl` command.
+
+Notable components:
+* The `openssl s_client` tool supports QUIC via the `-quic` flag 
+  (since version 3.2.0)
+* The `openssl s_server` tool does NOT support QUIC directly
+
+### crypto/
+
+The heart of OpenSSL containing implementations of cryptographic algorithms. 
+Inside you'll find subdirectories for different cryptographic functions. 
+This is the code that ends up in `libcrypto`.
+
+* `crypto/aes/` - AES encryption implementation
+* `crypto/bn/` - Big Number arithmetic library
+* `crypto/evp/` - High-level cryptographic API
+* `crypto/rsa/`, `crypto/ec/`, etc. - Various algorithm implementations
+
+### demos/
+
+Example code demonstrating how to use OpenSSL APIs:
+
+* `demos/quic/` - QUIC protocol examples, including a dedicated server 
+  implementation
+  * `demos/quic/server/` - Contains a standalone QUIC server application 
+    that should be used instead of `s_server` for QUIC functionality
+  * Usage: `./demos/quic/server/server <port> <certificate-file> <key-file>`
+
+### doc/
+
+Documentation organized in multiple formats:
+
+* `doc/man1/` - Manual pages for command-line tools
+* `doc/man3/` - Manual pages for API functions
+* `doc/man5/` - Manual pages for file formats
+* `doc/man7/` - Manual pages for overviews and concepts
+* `doc/internal/` - Developer documentation
+
+### include/
+
+Header files that are installed along with OpenSSL:
+
+* `include/openssl/` - Public API headers 
+* `include/crypto/` - Headers for internal cryptographic implementations
+* `include/internal/` - Headers for other internal components
+
+### providers/
+
+The provider architecture (new in OpenSSL 3.0) implementations:
+
+* `providers/default/` - The default provider
+* `providers/fips/` - The FIPS-validated module
+* `providers/legacy/` - Legacy algorithms
+
+### ssl/
+
+Implementation of SSL, TLS, DTLS, and QUIC protocols. 
+This is the code that ends up in `libssl`.
+
+* `ssl/record/` - SSL/TLS record layer
+* `ssl/statem/` - SSL/TLS state machine
+* `ssl/quic/` - QUIC protocol implementation (available since version 3.2.0)
+
+### test/
+
+Comprehensive test suite, organized by component and feature.
```

Comment:
Perhaps worthy of mention here is `test/recipes` where all the test recipes are held (it's these files that get invoked when you run `make test` - they then use the executables in the `test` directory. Also worthy of mention is `test/certs`